### PR TITLE
Add parameters to make custom backbone for detr

### LINKED ISSUE #14875: Add `in_chans` to `DetrModel`
# 🚀 Feature request

Add `in_chans` parameter to `DetrConfig` and corresponding models

## Motivation
I find myself wanting to test DETR on data that have only one channel (it can be grey scale images or spectrograms in my case). But to make it work I have to use hacks, that may eventually work purely.

## Your contribution
I will be happy to open PR with this if this is as useful as I see it. Maybe some other improvements that can be incorporated?

### PR DESCRIPTION
**Pull request by Zhylkaaa**
# What does this PR do?

Added few parameters to be able to create custom backbone models for different use cases (partially explained in mentioned issue)

Closes #14875  

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Maybe @LysandreJik can help with review?

### CODE REVIEW

File: src/transformers/models/detr/configuration_detr.py [hunk 1]
```diff
@@ -155,6 +155,10 @@ def __init__(
         bbox_loss_coefficient=5,
         giou_loss_coefficient=2,
         eos_coefficient=0.1,
+        in_chans=3,
```
Comment (NielsRogge) [inline src/transformers/models/detr/configuration_detr.py:158]:
```suggestion
        num_channels=3,
```
I would rename this to num_channels to be consistent with other models in the library (like ViT).

File: src/transformers/models/detr/configuration_detr.py [hunk 2]
```diff
@@ -155,6 +155,10 @@ def __init__(
...
+        pretrained=True,
```
Comment (NielsRogge) [inline src/transformers/models/detr/configuration_detr.py:159]: This can be renamed to `use_pretrained_backbone`, for clarity.

File: src/transformers/models/detr/configuration_detr.py [hunk 3]
```diff
@@ -155,6 +155,10 @@ def __init__(
...
+        freeze_layers=True,
```
Comment (NielsRogge) [inline src/transformers/models/detr/configuration_detr.py:160]: I would leave away this one, as there's already a method one can call on `DetrModel`, called `freeze_backbone` as seen [here](https://github.com/huggingface/transformers/blob/705ca7f21b2b557e0cfd5d0853b297fa53489d20/src/transformers/models/detr/modeling_detr.py#L1183).

Maybe we can improve its documentation for visibility.

File: src/transformers/models/detr/configuration_detr.py [hunk 4]
```diff
@@ -155,6 +155,10 @@ def __init__(
...
+        fix_batch_norm=True,
```
Comment (NielsRogge) [inline src/transformers/models/detr/configuration_detr.py:161]: What's the reason you want to replace the frozen batch norm layers?

File: src/transformers/models/detr/configuration_detr.py (hunk 3 above)
Comment (Zhylkaaa) [reply src/transformers/models/detr/configuration_detr.py:160]: I think this parameter is more specific, it disables freezing 2-4 layer of resnet-50 (this was just hard coded into encoder initialisation), but I can change it if you think freeze_backbone is better anyway

File: src/transformers/models/detr/configuration_detr.py (hunk 4 above)
Comment (Zhylkaaa) [reply src/transformers/models/detr/configuration_detr.py:161]: Training from scratch, I want to try training from fully randomly initialised model. Also I don't have pretrained backbone for my problem anyway, so I think this parameter won't harm
